### ISSUE TITLE
Error: vector when using tween_elements

### ISSUE THREAD

**Issue by marionowak** [thomasp85/tweenr#4]
I'm trying to interpolate x-y-data (named lire and koli in the data) with tweenr.

With this command:
`parlpos2 <- tween_elements(parlpos.tmp, time = 'jahr', group = 'id', ease = 'ease', nframes = 100)
` I am getting the error: `error: vector `

And when I use this command:
`parlpos2 <- tween_appear(data = parlpos.tmp, time = "jahr", nframes = 45)`
and insert the data.frame to ggplot with geom_path and use `frame = .frame` I am getting a full path like this, instead of something growing:
![output](https://cloud.githubusercontent.com/assets/9657746/14631443/b6ee88e4-0613-11e6-9bfd-72a5c6566fd5.gif)

ggplot:

```
p <-ggplot(parlpos2[parlpos2$.age+2014>0,], aes(x = lire, y = koli, group = rat, colour=rat, label=jahr, frame = .frame)) +
  geom_path(aes(cumulative = F))
```

The static data looks like this:
![screenshot 2016-04-19 09 40 48](https://cloud.githubusercontent.com/assets/9657746/14631250/d9ecb452-0612-11e6-8875-2dbb2659967a.png)

This is my data.frame `parlpos.tmp` :
```

parlpos <- read.xlsx("sr_nr14.xlsx", sheetName = "position")

parlpos$id = parlpos$jahr - 1980
parlpos$ease = 'linear'

parlpos.tmp <- parlpos

jahr rat  koli  lire id   ease
1  1981  NR 36.37  1.08  1 linear
2  1982  NR 35.55  1.29  2 linear
3  1983  NR 34.68  1.64  3 linear
4  1984  NR 33.60  1.82  4 linear
5  1985  NR 32.32  2.15  5 linear
6  1986  NR 30.74  2.53  6 linear
7  1987  NR 28.98  2.96  7 linear
8  1988  NR 27.22  3.37  8 linear
9  1989  NR 25.63  3.72  9 linear
10 1990  NR 24.38  4.02 10 linear
11 1991  NR 23.48  4.10 11 linear
12 1992  NR 22.86  4.18 12 linear
13 1993  NR 22.39  3.98 13 linear
14 1994  NR 22.01  3.84 14 linear
15 1995  NR 21.66  3.66 15 linear
16 1996  NR 21.26  3.48 16 linear
17 1997  NR 20.87  3.35 17 linear
18 1998  NR 20.45  3.40 18 linear
19 1999  NR 20.03  3.45 19 linear
20 2000  NR 19.59  3.64 20 linear
21 2001  NR 19.21  3.93 21 linear
22 2002  NR 18.85  4.33 22 linear
23 2003  NR 18.51  4.75 23 linear
24 2004  NR 18.17  5.17 24 linear
25 2005  NR 17.75  5.63 25 linear
26 2006  NR 17.39  6.00 26 linear
27 2007  NR 17.00  6.29 27 linear
28 2008  NR 16.66  6.51 28 linear
29 2009  NR 16.36  6.67 29 linear
30 2010  NR 16.11  6.74 30 linear
31 2011  NR 15.94  6.73 31 linear
32 2012  NR 15.87  6.70 32 linear
33 2013  NR 15.86  6.54 33 linear
34 2014  NR 16.00  6.45 34 linear
35 1981  SR 35.04 11.95  1 linear
36 1982  SR 34.51 12.28  2 linear
37 1983  SR 33.96 13.00  3 linear
38 1984  SR 33.67 13.16  4 linear
39 1985  SR 33.44 13.70  5 linear
40 1986  SR 33.40 14.31  6 linear
41 1987  SR 33.48 15.00  7 linear
42 1988  SR 33.74 15.61  8 linear
43 1989  SR 34.03 16.04  9 linear
44 1990  SR 34.32 16.48 10 linear
45 1991  SR 34.55 16.71 11 linear
46 1992  SR 34.64 16.89 12 linear
47 1993  SR 34.58 17.05 13 linear
48 1994  SR 34.52 17.39 14 linear
49 1995  SR 34.36 17.78 15 linear
50 1996  SR 34.17 18.20 16 linear
51 1997  SR 33.94 18.67 17 linear
52 1998  SR 33.78 19.15 18 linear
53 1999  SR 33.75 19.47 19 linear
54 2000  SR 33.81 19.85 20 linear
55 2001  SR 33.97 20.15 21 linear
56 2002  SR 34.15 20.30 22 linear
57 2003  SR 34.28 20.34 23 linear
58 2004  SR 34.38 20.18 24 linear
59 2005  SR 34.30 20.01 25 linear
60 2006  SR 34.12 19.45 26 linear
61 2007  SR 33.74 18.67 27 linear
62 2008  SR 33.21 17.69 28 linear
63 2009  SR 32.49 16.58 29 linear
64 2010  SR 31.67 15.37 30 linear
65 2011  SR 30.77 14.01 31 linear
66 2012  SR 29.86 12.72 32 linear
67 2013  SR 29.02 11.42 33 linear
68 2014  SR 28.35 10.15 34 linear
```

```
> str(parlpos.tmp)
'data.frame':	68 obs. of  6 variables:
 $ jahr: num  1981 1982 1983 1984 1985 ...
 $ rat : Factor w/ 2 levels "NR","SR": 1 1 1 1 1 1 1 1 1 1 ...
 $ koli: num  36.4 35.5 34.7 33.6 32.3 ...
 $ lire: num  1.08 1.29 1.64 1.82 2.15 2.53 2.96 3.37 3.72 4.02 ...
 $ id  : num  1 2 3 4 5 6 7 8 9 10 ...
 $ ease: chr  "linear" "linear" "linear" "linear" ...
```

data in Excel: [sr_nr14.xlsx](https://github.com/thomasp85/tweenr/files/225490/sr_nr14.xlsx)

Any idea what is going on?

**Comment by thomasp85** [comment]
I'll be happy to look into it once I've handed in my PhD in a couple of weeks - I'm afraid this is taking up all my time right now. I hope you can wait for that...

Quick thought: The error is from Rcpp and indicates that a wrong data structure has been passed in, so it's probably a matter of some formatting...

**Comment by marionowak** [comment]
sure - good luck with your PhD!
and btw: tweenr looks really promising!

**Comment by biologicaldynamics** [comment]
I had the same error, so I tried to reproduce the problem starting with the tween_elements example. I think the issue is the factor column. Try dropping that and run it again.

**Comment by biologicaldynamics** [comment]
I think the "time" column must be called "time" or else you get this error. Try renaming "jahr" to "time" and running it again.

Here is how I diagnosed this problem in my own data. I first filtered my dataset so that it was 100 observations, as with the sample data. I dropped all columns in the sample data except "x", "t", "ease" and "id". Then I put my filtered dataset in the same format, renaming columns so that the two tables were identical except for the actual values. They both worked fine.

Then I started changing the column names back to my original column names, one at a time. I was able to change "id" to "bay.id", "x" to "bounce" without problems. I could even rename "ease" to "easy." But if I renamed "time" back to "t," I got `Error: vector.`

```
reference <- df %>%
  arrange(id, time) %>%
  select(id, time, x, ease)

good <- my.data %>%
  filter(t <= 10) %>%
  filter(bay.id <= 12) %>%
  arrange(bay.id, t) %>%
  as.data.frame() %>%
  select(id = bay.id, time = t, bounce, easy = ease)

bad <- my.data %>%
  filter(t <= 10) %>%
  filter(bay.id <= 12) %>%
  arrange(bay.id, t) %>%
  as.data.frame() %>%
  select(id = bay.id, t, x = bounce, ease = ease)

ref.tween  <- tween_elements(reference, 'time', 'id', 'ease', nframes = 500)
good.tween <- tween_elements(good, 'time', 'id', 'easy', nframes = 500)
bad.tween  <- tween_elements(bad, "t", "id", "ease", nframes = 200)
```

**Comment by thomasp85** [comment]
This is a stupid oversight on my part - will soon be fixed

**Comment by thomasp85** [comment]
Fixed in 5c4383927d37916519c4f946a8af7a028e6318d1
